File: app/MultiMarket/c/src/server/acc.c
// Copyright 23-Jun-2019 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

#include "server/acc.h"
#include "server/acc/acc__downloader.h"
#include "server/acc/acc__companies.h"
#include "server/acc/acc__balance.h"
#include "server/acc/acc__trading.h"
#include "server/acc/acc__profits.h"

char *acc_process(AsyncActor *ac, Map *mrq) {
  CGI_GET_STR(source, mrq)

  if (str_eq(source, "Downloader")) return acc__downloader_process(ac, mrq);
  if (str_eq(source, "Companies")) return acc__companies_process(ac, mrq);
  if (str_eq(source, "Balance")) return acc__balance_process(ac, mrq);
  if (str_eq(source, "Trading")) return acc__trading_process(ac, mrq);
  if (str_eq(source, "Profits")) return acc__profits_process(ac, mrq);

  EXC_ILLEGAL_ARGUMENT(
    "source",
    "Downloader | Companies | Balance | Trading | Profits",
    source
  )
  return NULL; // Unreachable
}

File: app/KtMarket/go/server/pgs/acc/acc.go
// Copyright 19-Jun-2022 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Accounting data group hub.
package acc

import (
	"github.com/dedeme/KtMarket/server/pgs/acc/balance"
	"github.com/dedeme/KtMarket/server/pgs/acc/companies"
	"github.com/dedeme/KtMarket/server/pgs/acc/jail"
	"github.com/dedeme/KtMarket/server/pgs/acc/profits"
	"github.com/dedeme/KtMarket/server/pgs/acc/speedometers"
	"github.com/dedeme/KtMarket/server/pgs/acc/trading"
	"github.com/dedeme/ktlib/cgi"
	"github.com/dedeme/ktlib/js"
)

func Process(ck string, rq cgi.T) string {
	source := js.Rs(rq["source"])
	switch source {
	case "companies":
		return companies.Process(ck, rq)
	case "balance":
		return balance.Process(ck, rq)
	case "trading":
		return trading.Process(ck, rq)
	case "profits":
		return profits.Process(ck, rq)
	case "speedometers":
		return speedometers.Process(ck, rq)
	case "jail":
		return jail.Process(ck, rq)
	default:
		panic("Value of source (" + source + ") is not valid")
	}
}
